docs/site/migration/components/project-layout.md

```diff
@@ -0,0 +1,131 @@
+[Using components](../../Using-components.md), a typical LoopBack 4 component is
+an npm package exporting a Component class.
+
+The component class is usually implemented inside
+`src/{component-name}.component.ts` file, for example
+[src/metrics.component.ts](https://github.com/strongloop/loopback-next/blob/38f10b240551227d2d030c2fe8ee206880c9e029/extensions/metrics/src/metrics.component.ts):
+
+```ts
+import {Application, Component, CoreBindings} from '@loopback/core';
+import {bind, config, ContextTags, inject} from '@loopback/context';
+import {MetricsBindings} from './keys';
+import {DEFAULT_METRICS_OPTIONS, MetricsOptions} from './types';
+
+@bind({tags: {[ContextTags.KEY]: MetricsBindings.COMPONENT}})
+export class MetricsComponent implements Component {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE)
+    private application: Application,
+    @config()
+    options: MetricsOptions = DEFAULT_METRICS_OPTIONS,
+  ) {
+    // ...
+  }
+  // ...
+}
+```
+
+The code snippet above also shows how a LoopBack 4 component can receive
+configuration and the target application object.
+
+## Usage instructions
+
+LoopBack 4 components are added to applications inside application constructor.
+
+First, the application file needs to import the component class:
+
+```ts
+import {MetricsComponent} from '@loopback/extension-metrics';
+```
+
+Then in the constructor, add the component to your application:
+
+```ts
+this.component(MetricsComponent);
+```
+
+Finally, the application can configure the component by adding the following
+code to its constructor:
+
+```ts
+this.configure(MetricsBindings.COMPONENT).to({
+  // the configuration
+});
+```
+
+## Migration steps
+
+It is not feasible to migrate a LoopBack 3 component project to a LoopBack 4
+component project in a series of incremental changes done within the same
+repository. We recommend to create a new project using
+[`lb4 extension`](../../Extension-generator.md) CLI command and then
+incrementally migrate artifacts from the original LoopBack 3 component to the
+new project.
+
+<!--
+FIXME: create a follow-up task to update Extension template to follow
+the latest style as used by `extensions/metrics`:
+- inject the target app and component config to the component class,
+- provide `src/keys.ts` exporting `{ComponentName}Keys.COMPONENT` binding key,
+- decorate the component class with `@bind()`,
+- include generic Basic Use instructions in the scaffolded README file.
+As part of this task, we should also revisit the pages "Creating components"
+and "Using components", to make sure they are consistent with this migration
+guide and the updated generator (project template). For example, the pages
+should explain how to receive and provide component config via `@config` and
+`this.configure()` API, and so on.
+
+When the issue is created, then add a "note" paragraph here with a link to the
+new issue and a short explanation that users must make those changes manually
+for now.
+-->
```

Comment:
I'll replace this FIXME comment with a note pointing to the GitHub issue I'll create after we agree this is the direction to take.

Reply:
Created https://github.com/strongloop/loopback-next/issues/5336